tests/functional/test_config_groups.py

```diff
@@ -28,64 +28,86 @@
-            test_host_2 = provider.host_create(fqdn=self.SECOND_HOST)
-        return test_host_1, test_host_2
-
-    @allure.step('Check error')
-    def assert_that_host_add_is_unavailable(self, service_group: GroupConfig, host: Host):
-        with allure.step(f"Check that error is '{GROUP_CONFIG_HOST_ERROR.code}'"):
-            with pytest.raises(ErrorMessage) as e:
-                service_group.host_add(host)
-            GROUP_CONFIG_HOST_ERROR.equal(e)
-        with allure.step(f"Check error message is '{self.HOST_ERROR_MESSAGE}'"):
-            assert (
-                self.HOST_ERROR_MESSAGE in e.value.error['desc']
-            ), f"Should be error message '{self.HOST_ERROR_MESSAGE}'"
-
-    @allure.step("Check that the only second host is present in candidates on second provider group")
-    def assert_host_candidate_equal_expected(self, group: HostList, expected_hosts_amount: int):
-        with allure.step(f"Check that {expected_hosts_amount} hosts are available in group"):
-            assert len(group) == expected_hosts_amount, f"{expected_hosts_amount} hosts should be available in group"
-        with allure.step(f"Check that host '{self.SECOND_HOST}' is available in group"):
-            assert group[0].fqdn == self.SECOND_HOST, f"Should be available host '{self.SECOND_HOST}'"
+@allure.step('Check error')
+def assert_that_host_add_is_unavailable(service_group: GroupConfig, host: Host):
+    with allure.step(f"Check that error is '{GROUP_CONFIG_HOST_ERROR.code}'"):
+        with pytest.raises(ErrorMessage) as e:
+            service_group.host_add(host)
+        GROUP_CONFIG_HOST_ERROR.equal(e)
+    with allure.step(f"Check error message is '{HOST_ERROR_MESSAGE}'"):
+        assert HOST_ERROR_MESSAGE in e.value.error['desc'], f"Should be error message '{HOST_ERROR_MESSAGE}'"
+
+
+@allure.step('Check that host exists')
+def assert_that_host_exists(service_group: GroupConfig, host: Host):
+    with allure.step(f"Check that error is '{GROUP_CONFIG_HOST_EXISTS.code}'"):
+        with pytest.raises(ErrorMessage) as e:
+            service_group.host_add(host)
+        GROUP_CONFIG_HOST_EXISTS.equal(e)
+    with allure.step(f"Check error message is '{HOST_EXISTS_MESSAGE}'"):
+        assert HOST_EXISTS_MESSAGE in e.value.error['desc'], f"Should be error message '{HOST_EXISTS_MESSAGE}'"
+
 
+@allure.step("Check that the only second host is present in candidates on second provider group")
+def assert_host_candidate_equal_expected(group: HostList, expected_hosts_name: [str]):
+    with allure.step(f"Check that {len(expected_hosts_name)} hosts are available in group"):
+        assert len(group) == len(expected_hosts_name), f"{len(expected_hosts_name)} hosts should be available in group"
+    with allure.step(f"Check that host '{SECOND_HOST}' is available in group"):
+        assert [g.fqdn for g in group] == expected_hosts_name, f"Should be available hosts '{expected_hosts_name}'"
```

Comment:
bind `len(expected_hosts_name)` earlier and use it here and later

Reply:
I dont need len there, this line checks names of hosts

Reply:
You use `len(expected_hosts_name)` 3 times here, I've just selected more lines  that needed

Reply:
added variable